src/lib/authentication/arkBaseQuery.ts

```diff
@@ -0,0 +1,53 @@
+import { fetchBaseQuery } from "@reduxjs/toolkit/query";
+import type {
+  BaseQueryFn,
+  FetchArgs,
+  FetchBaseQueryError,
+} from "@reduxjs/toolkit/query";
+import { authProvider } from "../../features/authentication/authenticationSlice";
+import {
+  tokenReceived,
+  loggedOut,
+} from "../../features/authentication/authenticationSlice";
+import { RootState } from "../../app/configureStore";
+
+function GetBaseQuery(baseUrl: string) {
+  return fetchBaseQuery({
+    baseUrl: baseUrl,
+    prepareHeaders: (headers, { getState }) => {
+      const token = (getState() as RootState).auth.data?.token;
```

Comment:
why discussed to use the AuthProvider as is needed to pass the baseUrl to know which tokens.

Reply:
we said in the discussion that this first iteration will not support that

Reply:
yes, but in the same way the interface support the audience, this code should support calling into it.

I'm fine with the provider implementation not implementing it.

Reply:
ok got it